onnxruntime/core/providers/cpu/controlflow/if.cc

```diff
@@ -90,15 +91,56 @@ class IfImpl {
   std::vector<std::pair<AllocationType, MLValue>> outputs_;
 };
 
+void If::CheckForPassthroughOptimization(const OpKernelInfo& info,
+                                         const ONNX_NAMESPACE::GraphProto& then_proto,
+                                         const ONNX_NAMESPACE::GraphProto& else_proto) {
+  // check if subgraph has a single identity node passing a single input through as a single output
+  auto check_identity_passthrough = [](const ONNX_NAMESPACE::GraphProto& proto) -> const std::string {
```

Comment:
>const [](start = 83, length = 5)

Would this prevent automatic move possibility on return value?

Reply:
In both returns we're creating an rvalue, so I wouldn't have thought so. However this method is only called when constructing the kernel and not during each execution, and a single std::string is involved in each call, so I don't think it matters.